feat: add openstack/compare page

samhotep opened the pull request:
## Done

- Created the `openstack/compare` page according to [figma](https://www.figma.com/design/Q8VkClYK41uoVziGIUyUqx/canonical.com-openstack---Sites?node-id=25-18272&p=f&t=RiAQksk7AeZcUjPv-0) design and [copydoc](https://docs.google.com/document/d/1_luFPxd5yZy6eJt3-GE7jX_G8m3wTjo-JDo1aUPOnpU/edit?tab=t.0).
- Added new form at `openstack/compare` , copy [here](https://docs.google.com/document/d/1BnyB-JgvGutdzTrem3pHZr5Lp8V-OMlNZUbtb3H2dAk/edit?tab=t.0)

## QA

- View the demo in your web browser at: http://0.0.0.0:8002/
- Run through the following [QA steps](https://discourse.canonical.com/t/qa-steps/152)
- Check that the copydoc updates are correct
- Check that the figma design is correct

## Issue / Card

Fixes [WD-24685](https://warthogs.atlassian.net/browse/WD-24685)

## Screenshots
<img width="500" height="2516" alt="localhost_8003_openstack_compare" src="https://github.com/user-attachments/assets/153af8f7-19fc-44b0-9a40-2a33902d3d34" />

eliman11 commented:
Thanks @samhotep, looks great! My comments -

**Page changes**
- [ ] Button in the hero should be in title case as it's a product name - "[Discover Canonical OpenStack](https://canonical.com/openstack/)"
- [ ] "Full-stack support subscription" item in the table under the "Other OpenStack distributions" column should have a cross
- [ ] "GPU / vGPU support" item under "Other OpenStack distributions" should have a tick
- [ ] Should have a dash (-) instead of an apostrophe in "Ubuntu Pro ' the most comprehensive.."
- [ ] Change to AE spelling: "skeptical" instead of "sceptical" in "Are you sceptical about OpenStack due to the overall complexity.."
- [ ] Missing Ubuntu Pro link in "..benefiting from long-term security and support under our [Ubuntu Pro](https://ubuntu.com/pro) subscription"
- [ ] "Check private cloud pricing calculator >" link should be https://ubuntu.com/openstack/pricing-calculator - I've corrected on the copydoc
- [ ] I believe the visual under "With a non-stop contribution since the very beginning" should be a video instead of a static image - Tagging @mattea-turic to check 
- [ ] The toggle function for the table doesn't work with JS disabled. Would we be able to hide the toggle and expose the table with the explanations by default when JS is disabled?
- [ ] Could we please add alt text for the values in the table? See [/landscape/compare](https://ubuntu.com/landscape/compare) for example. For the copy - checks = "Included", crosses = "Not included" and question marks = "Depends on the OpenStack distribution"
- [ ] Could you please remove alt text for the two stock images under "Move away from proprietary virtualization platforms" and "Repatriate your cloud workloads and regain control over your data" as they are purely decorative?
- [ ] We will also need to change the alt text for the video and chart, I'm asking Tytus and will let you know when I hear back!
- [ ] Without changing the actual styling of the text, could we please use h3 HTML tags for these subheadings so we don't skip levels in the heading hierarchy?
<img width="841" height="424" alt="Screenshot 2025-08-28 at 11 18 11" src="https://github.com/user-attachments/assets/c1fea65b-fc36-444c-9737-a09eb859245b" />

- [ ] Could we limit the width of the text column on the right to 4 columns on the grid?
<img width="853" height="372" alt="Screenshot 2025-08-28 at 11 19 19" src="https://github.com/user-attachments/assets/4ac8478d-4b6a-467b-9841-d416682471da" />

- [ ] On medium/small screens, I think it might be better to display the table as cards like we do on [/kubernetes/compare](https://ubuntu.com/kubernetes/compare) because of the amount of text in each row. What do you think @mattea-turic? Also the toggle seems to be hidden on mobile

**Form**
- [ ] Missing phone number field 
- [ ] Could we reverse the order of the fields such that the contact information appears before the "What would you like to talk with us about?" text field to follow the copydoc?
- [ ] We should also create a static page for the contact form under the URL https://canonical.com/openstack/compare/contact-us for users who have JS disabled? This should have the same form fields as the modal version of the form

mattea-turic commented:
Thanks @samhotep ! Looks good! Just a few notes:

**Globally:**
- [ ] Could you please use chevrons rather than "greater than" signs at the end of CTAs e.g. "Read the white paper - “From VMware to OpenStack” >" should be "Read the white paper - “From VMware to OpenStack” ›", and same for the link underneath i.e. "Check private cloud pricing calculator"

**"With a non-stop contribution..."**
- [ ] This has now been reworked into a spotlight section (see [Figma](https://www.figma.com/design/Q8VkClYK41uoVziGIUyUqx/canonical.com-openstack---Sites?node-id=219-18809&t=hrzzOxMx8vWFumAr-1))

**"OpenStack distributions"**
- [ ] There seems to be excessive bottom padding, but I can't really figure out why – not sure if it's because of double wrapping or something else. Could you check this pls?

**"Get in touch with Canonical"**
- [ ] Pls wrap this in a deep section

@eliman11 
> On medium/small screens, I think it might be better to display the table as cards

I agree! Let's try that please

samhotep commented:
Hey @mattea-turic @eliman11 thanks for the detailed review! 
I've addressed the comments and updated the spotlight section, please have a look when you can!

mattea-turic commented:
@samhotep 

Thanks for all the changes! Table looks great!

Just a couple more things:

**Move away from proprietary virtualization platforms:**
- [ ] The CTA here should be: "Read the white paper – “From VMware to OpenStack ›" (wrong chevron and using em-dash in demo)

**Repatriate your cloud workloads**
- [ ] The CTA here should be: "Check private cloud pricing calculator ›" (wrong chevron in demo)

**Globally:**
- [ ] There seems to be a problem at smaller break points, where there is a lot of whitespace:
<img width="675" height="630" alt="Screenshot 2025-08-29 at 16 52 26" src="https://github.com/user-attachments/assets/0e995bd7-44fb-4362-8ce3-047b13ea89d0" />

samhotep commented:
Thanks for the review, @mattea-turic!  I've addressed the issue - it looks like using `u-hide` adds some hidden styling to the width, it should be okay now on smaller screens

eliman11 commented:
LGTM, thanks for the edits @samhotep!

mattea-turic commented:
Looks great, thank youu!! @samhotep

samhotep commented on templates/openstack/compare/index.html:
```diff
@@ -0,0 +1,715 @@
+    "type": "description",
+    "item": {
+    "type": "html",
+    "content": '<p><a href="https://ubuntu.com/engage/openstack-ebook-beginners">Learn about OpenStack architecture and what makes Canonical OpenStack different from other OpenStack distributions.</a></p><br /><hr />'
```
Good idea

samhotep commented:
Thanks @muhammad-ali-pk, I've made the updates, please have a look again!

muhammad-ali-pk commented on templates/openstack/compare/index.html:
```diff
@@ -512,177 +547,184 @@ <h4 class="p-heading--3">lines of code contributed</h4>
+        }
+      },
+      {
+        "type": "description",
+        "item": {
+            "type": "html",
+            "content": "<br />"
+          }
+      },
```
use shallow padding for the item above

muhammad-ali-pk commented on templates/openstack/compare/index.html:
```diff
@@ -512,177 +547,184 @@ <h4 class="p-heading--3">lines of code contributed</h4>
+        }
+      },
+      {
+        "type": "description",
+        "item": {
+          "type": "html",
+          "content": "<br />"
+        }
+      },
```
use shallow padding for the item above

muhammad-ali-pk commented on templates/openstack/compare/index.html:
```diff
@@ -512,177 +547,184 @@ <h4 class="p-heading--3">lines of code contributed</h4>
+        "type": "description",
+        "padding": "shallow",
+        "item": {
+          "type": "text",
```
shouldn't the type here be `html`?

muhammad-ali-pk commented on templates/openstack/compare/index.html:
```diff
@@ -512,177 +547,184 @@ <h4 class="p-heading--3">lines of code contributed</h4>
+        "type": "description",
+        "padding": "shallow",
+        "item": {
+          "type": "text",
```
shouldn't the type here be `html`?
Similarly for other similar blocks below

muhammad-ali-pk commented on templates/openstack/compare/index.html:
```diff
@@ -512,177 +547,184 @@ <h4 class="p-heading--3">lines of code contributed</h4>
+        "type": "description",
+        "padding": "shallow",
+        "item": {
+          "type": "text",
```
Type should be html

muhammad-ali-pk commented on templates/openstack/compare/index.html:
```diff
@@ -512,177 +547,184 @@ <h4 class="p-heading--3">lines of code contributed</h4>
+        "padding": "shallow",
+        "item": {
+          "type": "text",
+          "content": '<hr /><strong>Canonical OpenStack Datasheet</strong>'
```
The hr should be muted, as per design
Also for other similar blocks.

muhammad-ali-pk commented on templates/openstack/compare/index.html:
```diff
@@ -512,177 +547,184 @@ <h4 class="p-heading--3">lines of code contributed</h4>
+        "type": "description",
+        "item": {
+          "type": "text",
+          "content": '<strong>Openstack Ebook</strong>'
```
Ebook should be E-book
Couldn't comment it as suggestion

samhotep commented:
@muhammad-ali-pk I've updated the rules, please have another look 🙏🏿

muhammad-ali-pk commented:
<img width="877" height="608" alt="image" src="https://github.com/user-attachments/assets/72918c12-e6a1-453d-8aef-860739674035" />

The first two rules are still default rules. Please make them muted as well.
Also, please don't hide them on smalls screens.

samhotep commented on templates/openstack/compare/index.html:
```diff
@@ -0,0 +1,759 @@
+          }
+        }
+      },
+      {
+        "type": "description",
+        "padding": "shallow",
+        "item": {
+          "type": "html",
+          "content": ""
+        }
+      },
```
I see, changes coming up in a bit

samhotep commented:
@muhammad-ali-pk I've updated the rules to not hide on mobile, also they were muted already though I'm not sure why they weren't on the demo
<img width="1671" height="750" alt="image" src="https://github.com/user-attachments/assets/31af131a-fcd1-45ea-a09c-708d55ed1c93" />

muhammad-ali-pk commented on templates/openstack/compare/index.html:
```diff
@@ -0,0 +1,745 @@
+      },
+      {
+        "type": "description",
+        "padding": "shallow",
```
```suggestion
```
Not present in figma

muhammad-ali-pk commented on templates/openstack/compare/index.html:
```diff
@@ -0,0 +1,745 @@
+      },
+      {
+        "type": "description",
+        "padding": "shallow",
```
```suggestion
```

muhammad-ali-pk reviewed (approved):
LGTM! Just one minor change
